### PR TITLE
Type hint message interface

### LINKED ISSUE #49: How to persist custom events with PES
Provide an example and may improve the ES API to ease usage of custom domain events.

### LINKED ISSUE #51: Inject MessageFactory and Converter via adapter options
The event store Configuration should pass a `Prooph\Common\Messaging\MessageFactory` and a `Prooph\Common\Messaging\MessageConverter` as options to the event store adapter.
If no custom implementations were given the default implementations provided by prooph/common should be used.

### PR DESCRIPTION
**Pull request by codeliner**
- Closes #49 
- Closes #51

### CODE REVIEW

**Comment by codeliner** [comment]
ping @prolic for CR

File: tests/Mock/AdapterMock.php [hunk 1]
```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * This file is part of VehicleManagement
+ * Date: 8/13/15 - 11:34 PM
+ * (c) Sixt GmbH & Co. Autovermietung KG
```
Comment (prolic) [inline tests/Mock/AdapterMock.php]: change this, please.

File: tests/Mock/AdapterMock.php [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+     * @param array $metadata If empty array is provided, then all events should be returned
+     * @param null|int $minVersion Minimum version an event should have
+     * @return Message[]
+     */
+    public function loadEventsByMetadataFrom(StreamName $streamName, array $metadata, $minVersion = null)
+    {
+        throw new \BadMethodCallException(__CLASS__ . '::' . __METHOD__);
+    }
+}
+
```
Comment (prolic) [inline tests/Mock/AdapterMock.php]: missing new line at end of file

**Comment by prolic** [comment]
just two minor remarks, good to merge

File: tests/Mock/AdapterMock.php (hunk 1 above)
Comment (codeliner) [inline tests/Mock/AdapterMock.php]: Good catch. Crazy, my phpstorm project class template switch did not have a rule for the event-store project. Fortunately, only the two new added class files have the wrong header. I'll change that and align my template set up.

**Pull request merged**